var/spack/repos/builtin/packages/r-nloptr/package.py

```diff
@@ -42,3 +42,15 @@ class RNloptr(RPackage):
     version('1.0.4', 'f2775dfb4f7f5552d46937a04c062b0d')
 
     depends_on('r-testthat', type=('build', 'run'))
+    depends_on('nlopt')
+
+    def configure_args(self):
+        nlopt_spec = self.spec['nlopt']
+        nlopt_include = nlopt_spec.prefix.include
+        nlopt_lib = find_libraries('libnlopt', nlopt_spec.prefix,
+                                   recursive=True)
+        args = [
+            '--with-nlopt-cflags=-I{0}'.format(nlopt_include),
+            '--with-nlopt-libs={0}'.format(nlopt_lib),
+        ]
```

Comment:
Can you try the following and see if it works?
```python
args = [
    '--with-nlopt-cflags={0}'.format(self.spec['nlopt'].headers.include_flags),
    '--with-nlopt-libs={0}'.format(self.spec['nlopt'].libs.libraries[0])
]
```
The `.libs` and `.headers` attributes are designed so that packages can specify where to find their own headers/libraries instead of dozens of other packages guessing where they live.

Reply:
#3377 will add some user docs on this, but the API docs are probably more useful for the nitty gritty details. See https://github.com/spack/spack/issues/8997#issuecomment-413637793 for a list of where to find API docs for both `.libs` and `.headers` and friends, as well as their default implementations.